### ISSUE TITLE
Feature request: Lifecycle Management for Amazon EBS Snapshots

### ISSUE THREAD

**Issue by maciejdrozdzowski** [terraform-providers/terraform-provider-aws#5176]
As per https://aws.amazon.com/blogs/aws/new-lifecycle-management-for-amazon-ebs-snapshots/.

**Comment by bflad** [comment]
Dependency pull request submitted: #5211 

SDK Reference: https://docs.aws.amazon.com/sdk-for-go/api/service/dlm/

It looks like this can be a single new resource, something like:

```hcl
# Implementation may change during development
resource "aws_dlm_lifecycle_policy" "example" {
  description = ""
  executution_role_arn = ""
  state = ""

  policy_details {
    resource_types = []
    target_tag = {}

    schedule {
      name = ""

      create_rule {
        interval = #
        interval_unit = ""
      }

      retain_rule {
        count = #
      }
    }
  }
}
```

**Comment by tomelliff** [comment]
If I get some spare time in the next couple of days I'm going to start on this so I can replace my slightly janky Lambda functions that snapshot EBS volumes across our accounts.

**Comment by sagyp** [comment]
Is there any update on this feature?

**Comment by tomelliff** [comment]
Sorry for the delay, company moved office so I've had a busy couple of weeks recently. I started hacking on the resource this morning but I'm seeing a date parsing error from the SDK that is confusing me right now.

Once I work out what's wrong with that then I'll sort updates, fix a bunch of TODO's, generally clear things up then add tests and documentation. Very early WIP is viewable at https://github.com/terraform-providers/terraform-provider-aws/compare/master...tomelliff:dlm-lifecycle-policy-resource if anyone wants to track progress before I raise a pull request.

If anyone happens to know if there's a way I can handle the SDK attempting to parse the created/modified timestamps wrong (see https://github.com/aws/aws-sdk-go/issues/2100) then it would be good if they could point me in the right direction. Right now it seems like the SDK is wrong but that surprises me and I suspect I must be doing something wrong instead although I'm not sure what.

**Comment by bflad** [comment]
Support for managing DLM Lifecycle Policies via a new `aws_dlm_lifecycle_policy` resource has been merged and will release with version 1.43.0 of the AWS provider, likely later today or tomorrow.

Special shout out to @tomelliff for going above and beyond to work with AWS to get the blocking issue resolved! 🚀

**Comment by bflad** [comment]
This has been released in [version 1.43.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1430-november-07-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
